cargo-pgx/src/command/schema.rs

```diff
@@ -390,10 +390,14 @@ pub(crate) fn generate_schema(
             .expect(&format!("Couldn't call __pgx_sql_mappings"));
         sql_mapping = sql_mappings_symbol();
 
-        let symbol: libloading::os::unix::Symbol<unsafe extern "C" fn() -> SqlGraphEntity> = lib
+        let symbol: libloading::os::unix::Symbol<
+            unsafe extern "C" fn() -> eyre::Result<ControlFile>,
+        > = lib
```

Comment:
It seems unlikely a Result returning function like this is actually FFI-safe, `extern "C" fn` or no. Rust enums are usually not FFI-safe except in some qualified cases. This probably works but it is probably still UB, unfortunately. Maybe this should be using `extern "Rust" fn`?

Reply:
That does look like UB to me. That issue also affects the other two external function calls in this file (to `__pgx_sql_mappings` and `__pgx_internals*`), which all return `repr(Rust)` `enum`s.

I think even a `extern "Rust" fn` would still technically be UB since the compiler doesn't guarantee that the representation of an item is the same across compilation sessions (although in practice it usually is, at least when the target and compiler version are the same). Serializing data to a C array as it goes across the FFI boundary would remove that UB (although doing so would probably result in a minor hit to performance).

Reply:
I agree the guarantees are very thin here, but a `repr(Rust)` enum should have a consistent result, in practice, with the same _compiler_ (there are various reasons why this _has_ to be true), and is less likely to be a bear (on performance or otherwise).

Reply:
I updated the PR to make the relevant functions `extern "Rust"`. `extern "Rust"` is implied if nothing is specified but I wrote it out explicitly on the relevant functions for clarity.

I also documented that the `cargo-pgx` compiler should match the compiler used to compile crates that use it.

Reply:
It turns out rustfmt doesn't like explicit `extern "Rust"`, so I've made them implict. (rustfmt only changed the ones that weren't in a macro but I did all of them for consistency)

Reply:
I think `#[rustfmt::skip]` is better here.

Reply:
Updated to use `#[rustfmt::skip]`.